### ISSUE TITLE
[BUG] `hyperfine -h` panics when stdout is closed

### ISSUE THREAD

**Issue by lopopolo** [sharkdp/hyperfine#453]
```console
$ RUST_BACKTRACE=1 hyperfine -h | false
thread 'main' panicked at 'Error writing Error to stdout: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', /Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/github.com-1ecc6299db9ec823/clap-2.33.3/src/errors.rs:401:55
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: core::result::Result<T,E>::expect
   4: clap::errors::Error::exit
   5: clap::app::App::get_matches_from::{{closure}}
   6: hyperfine::app::get_arg_matches
   7: hyperfine::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Clap provides a `get_matches_from_safe` API that returns a `Result` rather than panicking when failing to write to stdout.

Here's some MIT-licensed code I use in one of my projects that does this dance in a panic-free way:

https://github.com/artichoke/artichoke/blob/d527412f9438aeba4cadb1f4303237f6f9e0cd4d/src/bin/artichoke.rs#L138-L173

**Comment by lopopolo** [comment]
(I tried to file this bug on `fd` as well which has the same panicking behavior, but the GitHub Issues UI kept erroring for me)

**Comment by sharkdp** [comment]
Thank you for reporting this.

> (I tried to file this bug on `fd` as well which has the same panicking behavior, but the GitHub Issues UI kept erroring for me)

Oh, fantastic. Does this work https://github.com/sharkdp/fd/issues/new?

**Comment by lopopolo** [comment]
That did the trick:

- https://github.com/sharkdp/fd/issues/897

This appears to be a productive source of bugs across the Rust CLI ecosystem:

- https://twitter.com/artichokeruby/status/1464996910511452165
- https://www.reddit.com/r/rust/comments/r48hem/claps_defaults_cause_rust_clis_to_panic_on_help/

**Comment by lopopolo** [comment]
I've opened up this ticket with upstream clap to see if the maintainers are open to fixing this panic on the v2 branch:

- https://github.com/clap-rs/clap/discussions/3055

**Comment by sharkdp** [comment]
Fixed via #459, if I understand correctly.
